src/mindlessgen/molecules/miscellaneous.py

```diff
@@ -81,6 +85,32 @@ def set_random_charge(ati: np.ndarray, verbosity: int = 1) -> tuple[int, int]:
         return charge, uhf
 
 
+def calculate_protons(natoms: np.ndarray) -> int:
+    """
+    Calculate the number of protons in a molecule from the atom list.
+    """
+    protons = 0
+    for i, atom_count in enumerate(natoms):
+        protons += atom_count * (i + 1)
+    return protons
+
+
+def calculate_ligand_electrons(natoms: np.ndarray, nel: int) -> int:
+    """
+    Calculate the number of ligand electrons in a molecule if lanthanides or actinides are within the molecule.
+    """
+    f_electrons = sum(
+        occurence
+        * (
+            ati - 3 + 1
```

Comment:
```suggestion
            ati - 2
```
Explain origin of "-2 = -3 + 1" and where -3 and +1 comes from in the comment line